test/distributed/_tensor/test_dtensor_compile.py

```diff
@@ -557,11 +557,14 @@ def fn(x):
         self.assertExpectedInline(
             str(fw_graph_cell[0]).strip(),
             """\
-def forward(self, primals_1):
-    wait_tensor = torch.ops._c10d_functional.wait_tensor.default(primals_1)
-    sin = torch.ops.aten.sin.default(wait_tensor)
-    sin_1 = torch.ops.aten.sin.default(sin);  sin = None
-    return (sin_1, primals_1, wait_tensor)""",
+def forward(self, primals_1, primals_2):
+    with_effects = torch.ops.higher_order.with_effects(primals_1, torch.ops._c10d_functional.wait_tensor.default, primals_2);  primals_1 = None
+    getitem = with_effects[0]
+    getitem_1 = with_effects[1];  with_effects = None
```

Comment:
Noob q: why does this take and return two arguments? Which one is the wait target/result? What is the other one?

Reply:
The main idea of additional tokens is to wrap effectful ops into with_effects and add additional first argument that consumes "token" which is originally added to the graph inputs; with_effects will return (new_token, result), so with_effects[0] here is the next token, with_effects[1] is actual result:
```
def fn(x):
    return wait_tensor(x)
```
```
forward(token, x)
    new_token, y = with_effects(token, wait_tensor, x)
return new_token, y
```

Reply:
Thanks for the explanation! Just to verify my understanding, each graph will have at most one effect token as argument, which "chains" all the effectful ops?

Reply:
that's right (in theory, you could have multiple effect token arguments to the graph, so you can create separate "dependency chains", if you decide that having on giant dependency chain for all effectful ops is too restrictive. Multiple tokens is not yet implemented though)